base64.js

```diff
@@ -1,6 +1,6 @@
 ;(function () {
 
-  var object = typeof exports != 'undefined' ? exports : self; // #8: web workers
+  var object = typeof exports != 'undefined' ? exports : typeof self != 'undefined' ? self : (1,eval)('this'); // #8: web workers, #31: ExtendScript
```

Comment:
Does this work for you? For me,

```javascript
(function() { return (1, eval)('this'); }())
```

evaluates to a reference to the anonymous function. On the other hand,

```javascript
(function(global) { return global; }(this))
```

evaluates to the global object.

Now that we have three cases we could reformat the ternary expressions to improve readability. I suggest something along these lines:

```javascript
var object =
  typeof exports != 'undefined' ? exports :
  typeof self != 'undefined'    ? self :
  /* otherwise */                 ...;
```